packages/jest-worker/src/workers/messageParent.ts

```diff
@@ -6,32 +6,35 @@
  */
 
 import {PARENT_MESSAGE_CUSTOM} from '../types';
+import {serialize} from './utils';
 
-const isWorkerThread = () => {
+const isWorkerThread: boolean = (() => {
   try {
     // `Require` here to support Node v10
-    const {isMainThread, parentPort} = require('worker_threads');
-    return !isMainThread && parentPort;
+    const {
+      isMainThread,
+      parentPort,
+    } = require('worker_threads') as typeof import('worker_threads');
+    return !isMainThread && parentPort != null;
   } catch {
     return false;
   }
-};
+})();
 
-const messageParent = (
+export default function messageParent(
   message: unknown,
-  parentProcess: NodeJS.Process = process,
-): void => {
-  try {
-    if (isWorkerThread()) {
-      // `Require` here to support Node v10
-      const {parentPort} = require('worker_threads');
-      parentPort.postMessage([PARENT_MESSAGE_CUSTOM, message]);
-    } else if (typeof parentProcess.send === 'function') {
-      parentProcess.send([PARENT_MESSAGE_CUSTOM, message]);
-    }
-  } catch {
```

Comment:
this catch swallowed the error - it should instead just allow any errors to bubble up

Reply:
landed in #10984